accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,191 @@
+
+Standalone source-generator NuGet packages may be used for components which are in the shared framework when the source-generator is not ready to stabilize at the same time as the shared framework, and the component inside the shared framework does not need to be changed.  If the component in the shared framework needs to be changed, it should be done so by either a library+generator NuGet package (`System.Text.Json`) or an experimental build of the entire framework, typically from a codebase like [dotnet/runtimelab](https://github.com/dotnet/runtimelab).
+
+###  I really really need to execute runtime code on the TargetFramework!
+
+You probably need a designer or independent build component that has a host process that can leverage the runtime targeting information of the application.  At the moment this doesn't have a great pattern to follow with source generators.  The compiler does not provide a host, nor does it know anything about target runtime (or in some cases executable frameworks).
+
+### How does this impact source build?
+
+For Linux distribution .NET defines a process from which the .NET product can be built from source: [dotnet/source-build](https://github.com/dotnet/source-build).  This process requires building the entire .NET product without any pre-built binaries.  Some components which make up .NET target older frameworks which aren't built from the latest codebase.  This is done because these components may need to run at design-time on a different framework (.NETFramework), or may also ship out-of-band as a NuGet package for use by libraries that work on many frameworks.  To support building these components on past frameworks, source-build uses an alternate representation of reference assets for those frameworks: [dotnet/source-build-reference-packages](https://github.com/dotnet/source-build-reference-packages).  In order to satisfy build-from-source packages should only contain reference-API and metadata, no executable code.  As a result, the source-build-reference-packages that represent past framework versions will omit source generators.  This could be a problem for any component which needs to build during source-build and targets this old framework, for example a project targeting .NET 6.0 in the .NET 7.0 source build.  That component would fail to build during source build due to the missing source-generator (see [Deterministic failure](#Deterministic%20failure)).  That component can mitigate the failure by either including a .NET 7.0 configuration for source build.  Alternatively, if the dependent source-generator is also out-of-band (for example: System.Text.Json) the failing component may reference the latest dependent out-of-band package to get the latest source-generator from the package instead of the ref-pack.  To force components to make this mitigation early, we will make sure that Arcade disables source-generators from ref-packs which are not the latest when building for source build.
```

Comment:
```suggestion
For Linux distribution .NET defines a process from which the .NET product can be built from source: [dotnet/source-build](https://github.com/dotnet/source-build).  This process requires building the entire .NET product without any pre-built binaries.  Some components which make up .NET target older frameworks which aren't built from the latest codebase.  This is done because these components may need to run at design-time on a different framework (.NETFramework), or may also ship out-of-band as a NuGet package for use by libraries that work on many frameworks.  To support building these components on past frameworks, source-build uses an alternate representation of reference assets for those frameworks: [dotnet/source-build-reference-packages](https://github.com/dotnet/source-build-reference-packages).  In order to satisfy build-from-source, packages should only contain reference-API and metadata, no executable code.  As a result, the source-build-reference-packages that represent past framework versions will omit source generators.  This could be a problem for any component which needs to build during source-build and targets this old framework, for example a project targeting .NET 6.0 in the .NET 7.0 source build.  That component would fail to build during source build due to the missing source-generator (see [Deterministic failure](#Deterministic%20failure)).  That component can mitigate the failure by either including a .NET 7.0 configuration for source build.  Alternatively, if the dependent source-generator is also out-of-band (for example: System.Text.Json) the failing component may reference the latest dependent out-of-band package to get the latest source-generator from the package instead of the ref-pack.  To force components to make this mitigation early, we will make sure that Arcade disables source-generators from ref-packs which are not the latest when building for source build.
```